plugins/in_kafka/in_kafka.c

```diff
@@ -243,6 +246,21 @@ static int in_kafka_init(struct flb_input_instance *ins,
         goto init_error;
     }
 
+    /* Set the kafka poll timeout dependend on wether we run in our own 
+     * or in the main event thread. 
+     * a) run in main event thread: 
+     *    -> minimize the delay we might create
+     * b) run in our own thread: 
+     *    -> optimize for throuput and relay on 'fetch.wait.max.ms' 
+     *       which is set to 500 by default default. lets set it to
+     *       twice that so that increasing fetch.wait.max.ms still
+     *       has an effect.
+     */
+    ctx->poll_timeount_ms = 1;   
+    if(ins->is_threaded) {
+        ctx->poll_timeount_ms = 1000;
```

Comment:
Wondering if `poll_timeout_ms` can be exposed as a config option? Given that `rdkafka.fetch.wait.max.ms` can be modified to fit the needs.